eth_utils/numeric.py

```diff
@@ -0,0 +1,14 @@
+import numbers
+from typing import TypeVar
+
+TValue = TypeVar("TValue", bound=numbers.Real)
```

Comment:
I think we want to accept any (matching) comparable values, something like:

```py
from typing import TypeVar
from typing_extensions import Protocol

CT = TypeVar('CT', bound="Comparable")

class Comparable(Protocol):
    @abstractmethod
    def __lt__(self, other: CT) -> bool: ...
```

Untested, inspired by:
- https://stackoverflow.com/questions/37669222/how-can-i-hint-that-a-type-is-comparable-with-typing
- https://github.com/python/typing/issues/59

Looks like it may be not quite strict enough, though, based on this comment: https://github.com/python/typing/issues/59#issuecomment-354263666

Reply:
I don't think we can use `Protocols` in this library yet.  It's used too broadly across codebases that use older versions of pythong where it isn't yet available.

Reply:
Ah yeah. The SO example just used an `abc.ABC` (really the older `ABCMeta`), so maybe that's enough.

Reply:
OMG, that is so useful!